lib/tasks/config.rake

```diff
@@ -181,4 +181,8 @@ Rake::Task['db:seed'].enhance do
       Setting.create! setting.update(:category => "Setting::General")
     end
   end
+
+  File.open('/var/lib/foreman/database_seeded') do |file|
```

Comment:
This probably needs to be configurable since this will fail in development setups. I think we have a directory relative to the app_root with symlinks in place in production.

Reply:
How is this going to work on a multi host setup?

Reply:
Not very well I suppose. But if you are using the puppet modules on each host and updating packages on each host seems that they would all end up in the right state to database migrate and database seed. This is mostly targeting an attempt to simplify our packages, and keep the puppet modules idempotent.

I'm open to other ideas. Seeding is hard because they are idempotent in and of themselves and therefore there is no method to detect that they have ran like migrations.

Reply:
Another issue with storing it in the database is that multiple hosts might both be looking at the same value and decide they need to concurrently perform an action. That means you need to create a lock as well.